### PR TITLE
Fall back when no recommended course version

### PR DESCRIPTION
**Pull request by rshipp**
When there are no recommended course versions for a given selection, the section setup flow was raising an exception and not allowing certain courses to be assigned. Instead, we should fall back to the most recent stable course version. (Thanks Bethany for knowing what that behavior should be.)

image

## Links

- jira ticket: [TEACH-622](https://codedotorg.atlassian.net/browse/TEACH-622)

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx [hunk 1]
```diff
@@ -96,6 +96,14 @@ function updateSectionCourse(updateCourse, course) {
     )?.id;
   }
 
+  // If no recommended version, fall back to the most recent stable version.
+  if (courseVersionId === undefined) {
+    const stableVersions = Object.values(courseVersions).filter(
+      version => version.is_stable
+    );
+    courseVersionId = stableVersions[stableVersions.length - 1]?.id;
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx]: Do we know  for sure that these will always be in chronological order?

File: apps/test/unit/templates/sectionsRefresh/QuickAssignTableTest.jsx [hunk 2]
```diff
@@ -58,6 +59,21 @@ describe('QuickAssignTable', () => {
...
   });
 
+  it('correctly falls back when a course has no recommended version', () => {
+    const updateSpy = sinon.spy();
+    const wrapper = setUpMount({
+      updateCourse: updateSpy,
+      courseOfferings: noRecommendedVersionsOfferings,
+    });
+
+    const radio = wrapper.find('input');
```
Comment (hannahbergam) [inline apps/test/unit/templates/sectionsRefresh/QuickAssignTableTest.jsx:69]: Thanks for adding this test! The only thing I can think of would be to add one more stable version of a different year to ensure this is grabbing the most recent one.

**Review by hannahbergam — COMMENTED** [review]
This is super important work- thank you for cleaning it up!! Only two (related) comments.

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx (hunk 1 above)
Comment (rshipp) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx]: ah, no, that was just an assumption I made 😅 I'll check the api, and see if I need to sort these by something instead.

Comment (rshipp) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx]: the API isn't sorting, and just returns things in whatever order they come from sql. which should typically be oldest things first, but there's no guarantee. it looks like the `key` is consistently a "version year", so I'm sorting by that instead. unless you know of any exceptions to that rule?

**Review by hannahbergam — APPROVED** [review]
Thanks for adding this! I *think your instincts were right that it's sorted somewhere else, but I appreciate this explicitly residing here in case we ever change the display sorting

**Pull request merged**